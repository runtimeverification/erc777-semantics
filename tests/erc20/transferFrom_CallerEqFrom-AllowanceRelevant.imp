test.preamble();

// Testing transferFrom(From,To,Value) when From = caller.
// Allowance assigned specifically lower than Value.
// So the allowance is relevant in this case (unlike for transfer).

test.approve(caller, 20);   // set allowance of caller for itself below 23
test.transferFrom(caller, 4, 23);   // transfer 23 larger than allowance 20
