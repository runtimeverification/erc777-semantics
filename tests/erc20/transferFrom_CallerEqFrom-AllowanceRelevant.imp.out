7 is approving allowance 20 for 7
7 is transferring 23 from 7 to 4
REVERT
Balance of 7 is 70
Balance of 4 is 40
Allowance of 7 for 7 is 20
